lib/src/test/resources/test-cases.json

```diff
@@ -1847,12 +1931,30 @@
       {"tags": "hello"},
       false
     ],
+    [
+      "$all - fail null value",
+      {"tags": {"$all": ["one", "three"]}},
+      {"tags": null},
+      false
+    ],
     [
       "$nor - pass",
       {"$nor": [{"name": "john"}, {"age": {"$lt": 30}}]},
       {"name": "jim", "age": 40},
       true
     ],
+    [
+      "$nor - pass with null value for name",
+      {"$nor": [{"name": "john"}, {"age": {"$lt": 30}}]},
+      {"name": null, "age": 40},
+      true
+    ],
+    [
+      "$nor - pass with null value for age",
+      {"$nor": [{"name": "john"}, {"age": {"$lt": 30}}]},
+      {"name": "jim", "age": null},
+      true
```

Comment:
`null` is `0` in JavaScript.
`(0 < 30) = true`.
`false nor true = !(false or true) = false`